src/java.base/share/classes/sun/nio/ch/FileChannelImpl.java

```diff
@@ -556,21 +556,22 @@ private long transferToDirectly(long position, int icount,
             return IOStatus.UNSUPPORTED;
         int thisFDVal = IOUtil.fdVal(fd);
         int targetFDVal = IOUtil.fdVal(targetFD);
-        if (thisFDVal == targetFDVal) // Not supported on some configurations
+        if (thisFDVal != -1 && targetFDVal != -1 && thisFDVal == targetFDVal)
```

Comment:
How can the fd values be -1 here, we must have an issue somewhere else if this happens.

Reply:
On Windows this can happen.

Reply:
Removed in second commit.